### ISSUE TITLE
Random server disconnect

### ISSUE THREAD

**Issue by mrniddala** [TheCSDev/mc-item-despawn-flicker#1]
Description: Unexpected error

java.util.ConcurrentModificationException
	at Not Enough Crashes deobfuscated stack trace.(1.19.2+build.18)
	at java.util.HashMap$HashIterator.remove(HashMap.java:1611)
	at thecsdev.itemdespawnflicker.client.network.ClientPlayNetworkHandler.lambda$register$0(ClientPlayNetworkHandler.java:75)
	at net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents.lambda$static$6(ClientTickEvents.java:111)
	at net.minecraft.client.world.ClientWorld.handler$fmg000$tickWorldAfterBlockEntities(ClientWorld:8129)
	at net.minecraft.client.world.ClientWorld.tickEntities(ClientWorld:255)
	at net.minecraft.client.MinecraftClient.tick(MinecraftClient:1851)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1135)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:768)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.client.MinecraftClient.modify$imf000$atTheEndOfSecondCatchBeforePrintingCrashReport(MinecraftClient:32022)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:795)
	at net.minecraft.client.main.Main.main(Main:244)
	at net.minecraft.client.main.Main.main(Main:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

**Comment by TheCSDev** [comment]
Now that one I did not expect at all. (idk if you have any Java experience but) I never thought `Iterator.remove()` would throw a `ConcurrentModificationException`, of all things. I tried Googling why that one took place, but all I keep seeing is results about how that never happens at all, as literally the **whole purpose** of iterators is to **avoid** `ConcurrentModificationException`s.
I swear, this is part of why I don't like Java, as it sometimes acts in the most unpredictible ways possible. Kinda like crossing the street while making sure there are no cars at all, only to get hit by a plane.

With that said tho, thank you for reporting the bug. I will of course resolve it. In the meantime, I will also work on that other thing we mentioned on the CurseForge site.

**Comment by TheCSDev** [comment]
Thank you once again for the bug report. I have finally finished resolving it and implementing NBT Age change tracking. Now if a command block changes an item's age, that change will synchronize. And to avoid packet spam, I limited the synchronization frequency (while also making sure the latest change does eventually synchronize). It will take some time for CurseForge to approve the published update however. Have a great day once again.
